src/merkleblock.h

```diff
@@ -128,6 +128,10 @@ class CMerkleBlock
     /** Public only for unit testing */
     CBlockHeader header;
     CPartialMerkleTree txn;
+    /** Only used by subclass WtxidInBlockMerkleProof */
```

Comment:
There is no such subclass?

I think all this logic should entirely be in a separate class though. For efficiency probably two paths:

 * if the generation tx has no witness commitment; provde all (w)txids and the generation tx via the block partial merkle tree.
 * if the generation tx does have a witness commitment; provide just the generation tx's merkle path back to the block header, and prove all wtxids via a partial merkle tree back to the witness commitment